internal/formats/common/cyclonedxhelpers/format.go

```diff
@@ -0,0 +1,90 @@
+package cyclonedxhelpers
+
+import (
+	"time"
+
+	"github.com/CycloneDX/cyclonedx-go"
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/version"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/sbom"
+	"github.com/anchore/syft/syft/source"
+	"github.com/google/uuid"
+)
+
+func ToFormatModel(s sbom.SBOM) *cyclonedx.BOM {
+	cdxBOM := cyclonedx.NewBOM()
+	versionInfo := version.FromBuild()
+
+	cdxBOM.SerialNumber = uuid.New().URN()
+	cdxBOM.Metadata = toBomDescriptor(internal.ApplicationName, versionInfo.Version, s.Source)
+
+	packages := s.Artifacts.PackageCatalog.Sorted()
```

Comment:
Do we want this function to guarantee the sort, or should a user bring their own sorted packages already? This could prevent duplicate sort calls and not punish a consumer for having sorted previously in an upstream process.

Reply:
We do want to guarantee sorting here. Why? Overall we want to have the output documents be as stable as possible; the same bytes being encoded twice should result in the same serialized bytes. Since the package catalog is a map under the hood the order of packages is not guaranteed to be stable, so sorting is necessary in the format-model domain.